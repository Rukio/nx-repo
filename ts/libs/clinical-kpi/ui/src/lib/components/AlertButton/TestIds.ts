export const ALERT_BUTTON_TEST_IDS = {
  TEXT: 'alert-button-text',
  BUTTON: 'alert-button-button',
};
