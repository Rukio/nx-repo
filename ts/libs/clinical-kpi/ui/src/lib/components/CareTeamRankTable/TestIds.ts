export const CARE_TEAM_RANK_TABLE_TEST_IDS = {
  TABLE: 'care-team-rank-table',
  getRankTestId: (rank?: number) => `care-team-table-rank-${rank}`,
  getProviderNameTestId: (rank?: number) =>
    `care-team-rank-table-provider-name-${rank}`,
  getValueTestId: (rank?: number) => `care-team-rank-table-value-${rank}`,
  getValueChangeTestId: (rank?: number) =>
    `care-team-rank-table-value-change-${rank}`,
  getPlaceholderTestId: (rank?: number) =>
    `care-team-rank-table-placeholder-${rank}`,
};

export const CARE_TEAM_PLACEHOLDER_ROW_RANKS = [1, 2, 3, 4, 5, 6, 7];
