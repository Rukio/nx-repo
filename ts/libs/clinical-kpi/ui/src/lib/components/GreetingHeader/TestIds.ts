export const GREETING_HEADER_TEST_IDS = {
  GREETING_TEXT: 'header-greeting-text',
  LAST_UPDATED_TEXT: 'header-last-updated-text',
  VISITS_COMPLETED_TEXT: 'header-visits-completed-text',
  PERFORMANCE_TEXT: 'header-performance-text',
  LEARN_MORE_LINK: 'header-learn-more-link',
  LAST_UPDATED_PLACEHOLDER: 'header-last-updated-placeholder',
  VISITS_COMPLETED_PLACEHOLDER: 'header-visits-completed-placeholder',
  STATION_LINK: 'header-station-dashboard-link',
  STATION_LINK_LABEL_PREFIX: 'header-station-dashboard-link-label-prefix',
};
