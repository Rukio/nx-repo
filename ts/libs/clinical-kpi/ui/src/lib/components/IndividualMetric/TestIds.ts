export const INDIVIDUAL_METRIC_TEST_IDS = {
  TYPE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-type`,
  VALUE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-value`,
  VALUE_CHANGE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-change`,
  PERFORMANCE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-performance`,
  GOAL: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-goal`,
};
