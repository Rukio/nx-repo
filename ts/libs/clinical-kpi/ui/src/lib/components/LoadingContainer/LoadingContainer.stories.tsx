import { StoryFn, Meta } from '@storybook/react';
import { LoadingContainer } from './';

export default {
  component: LoadingContainer,
  title: 'LoadingContainer',
} as Meta<typeof LoadingContainer>;

const Template: StoryFn<typeof LoadingContainer> = (args) => (
  <LoadingContainer {...args} />
);

export const Primary = Template.bind({});
Primary.args = { testIdPrefix: 'storybook' };
