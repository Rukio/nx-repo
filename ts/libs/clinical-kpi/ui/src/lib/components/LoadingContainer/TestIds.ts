export const LOADING_CONTAINER_TEST_IDS = {
  SPINNER: (prefix: string) => `${prefix}-loading`,
};
