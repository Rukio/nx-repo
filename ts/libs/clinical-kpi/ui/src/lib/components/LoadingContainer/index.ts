export { default as LoadingContainer } from './LoadingContainer';
export { LOADING_CONTAINER_TEST_IDS } from './TestIds';
