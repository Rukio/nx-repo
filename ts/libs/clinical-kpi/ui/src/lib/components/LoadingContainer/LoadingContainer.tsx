import {
  Box,
  CircularProgress,
  CircularProgressProps,
  makeSxStyles,
} from '@*company-data-covered*/design-system';
import { LOADING_CONTAINER_TEST_IDS } from './TestIds';

/*
 * renders container with centered circular progress indicator
 * @param {String} prefix to prepend to data-testid -loading
 * @param {String} CSS container min-width value. Defaults to inherit
 * @param {String} CSS container min-height value. Defaults to inherit
 * @param {CircularProgressProps['size]} spinner size passed to MUI <CircularProgress>. Defaults to 100
 */
interface Props {
  testIdPrefix: string;
  cssWidth?: string;
  cssHeight?: string;
  spinnerSize?: CircularProgressProps['size'];
}

type StyleProps = Pick<Props, 'cssHeight' | 'cssWidth' | 'spinnerSize'>;

const makeStyles = ({ cssWidth, cssHeight, spinnerSize }: StyleProps) => {
  const halfSpinnerSize = spinnerSize && +spinnerSize / 2;

  return makeSxStyles({
    container: {
      position: 'absolute',
      top: '50%',
      left: '50%',
      marginTop: `-${halfSpinnerSize}px`,
      marginLeft: `-${halfSpinnerSize}px`,
      minWidth: cssWidth,
      minHeight: cssHeight,
    },
  });
};

const LoadingContainer = ({
  testIdPrefix,
  cssWidth = 'inherit',
  cssHeight = 'inherit',
  spinnerSize = 100,
}: Props) => {
  const styles = makeStyles({
    cssWidth,
    cssHeight,
    spinnerSize,
  });

  return (
    <Box sx={styles.container}>
      <CircularProgress
        size={spinnerSize}
        data-testid={LOADING_CONTAINER_TEST_IDS.SPINNER(testIdPrefix)}
      />
    </Box>
  );
};

export default LoadingContainer;
