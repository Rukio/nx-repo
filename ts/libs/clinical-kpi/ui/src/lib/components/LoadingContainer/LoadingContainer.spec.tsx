import { render, screen } from '@testing-library/react';

import LoadingContainer from './LoadingContainer';
import { LOADING_CONTAINER_TEST_IDS } from './TestIds';

describe('LoadingContainer', () => {
  it('should render correctly with default props (full container size)', () => {
    const testIdPrefix = 'full-screen';
    const { asFragment } = render(
      <div style={{ width: '100vw', height: '100vh' }}>
        <LoadingContainer testIdPrefix={testIdPrefix} />
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByTestId(LOADING_CONTAINER_TEST_IDS.SPINNER(testIdPrefix))
    ).toBeTruthy();
  });

  it('should render correctly with custom container size', () => {
    const testIdPrefix = 'pixel-sized';
    const { asFragment } = render(
      <div style={{ width: '90vw', height: '90vh' }}>
        <LoadingContainer
          testIdPrefix={testIdPrefix}
          cssWidth="300px"
          cssHeight="400px"
        />
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByTestId(LOADING_CONTAINER_TEST_IDS.SPINNER(testIdPrefix))
    ).toBeTruthy();
  });

  it('should render correctly with custom container and spinner size', () => {
    const testIdPrefix = 'spinner-sized';
    const { asFragment } = render(
      <div style={{ width: '90vw', height: '90vh' }}>
        <LoadingContainer
          testIdPrefix={testIdPrefix}
          cssWidth="300px"
          cssHeight="400px"
          spinnerSize={50}
        />
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByTestId(LOADING_CONTAINER_TEST_IDS.SPINNER(testIdPrefix))
    ).toBeTruthy();
  });
});
