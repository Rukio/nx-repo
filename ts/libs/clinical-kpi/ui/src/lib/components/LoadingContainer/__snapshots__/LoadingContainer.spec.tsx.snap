// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContainer should render correctly with custom container and spinner size 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -25px;
  margin-left: -25px;
  min-width: 300px;
  min-height: 400px;
}

.emotion-1 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-2 {
  display: block;
}

.emotion-3 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div
    style="width: 90vw; height: 90vh;"
  >
    <div
      class="MuiBox-root emotion-0"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-1"
        data-testid="spinner-sized-loading"
        role="progressbar"
        style="width: 50px; height: 50px;"
      >
        <svg
          class="MuiCircularProgress-svg emotion-2"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-3"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LoadingContainer should render correctly with custom container size 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -50px;
  margin-left: -50px;
  min-width: 300px;
  min-height: 400px;
}

.emotion-1 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-2 {
  display: block;
}

.emotion-3 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div
    style="width: 90vw; height: 90vh;"
  >
    <div
      class="MuiBox-root emotion-0"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-1"
        data-testid="pixel-sized-loading"
        role="progressbar"
        style="width: 100px; height: 100px;"
      >
        <svg
          class="MuiCircularProgress-svg emotion-2"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-3"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LoadingContainer should render correctly with default props (full container size) 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -50px;
  margin-left: -50px;
  min-width: inherit;
  min-height: inherit;
}

.emotion-1 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-2 {
  display: block;
}

.emotion-3 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div
    style="width: 100vw; height: 100vh;"
  >
    <div
      class="MuiBox-root emotion-0"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-1"
        data-testid="full-screen-loading"
        role="progressbar"
        style="width: 100px; height: 100px;"
      >
        <svg
          class="MuiCircularProgress-svg emotion-2"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-3"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
